crates/biome_js_analyze/src/aria_analyzers/nursery/use_valid_aria_role.rs

```diff
@@ -0,0 +1,117 @@
+use crate::aria_services::Aria;
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::jsx_ext::AnyJsxElement;
+use biome_rowan::AstNode;
+use serde::{Deserialize, Serialize};
+
+declare_rule! {
+    /// Elements with ARIA roles must use a valid, non-abstract ARIA role.
+    ///
+    /// Source: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/aria-role.md
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role="datepicker"></div>
+    ///  ```
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role="range"></div>
+    ///  ```
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role=""></div>
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    ///   <Foo role="foo"></Foo>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// <>
+    ///   <div role="button"></div>
+    ///   <div role={role}></div>
+    ///   <div></div>
+    /// </>
+    /// ```
+    ///
+    /// ## Accessibility guidelines
+    /// - [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)
+    ///
+    /// ## Resources
+    /// - [Chrome Audit Rules, AX_ARIA_01](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_01)
+    /// - [DPUB-ARIA roles](https://www.w3.org/TR/dpub-aria-1.0/)
+    /// - [MDN: Using ARIA: Roles, states, and properties](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques)
+    ///
+    pub(crate) UseValidAriaRole {
+        version: "next",
+        name: "useValidAriaRole",
+        recommended: false,
+    }
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct UseValidAriaRoleOptions {
+    allowed_invalid_roles: Vec<String>,
+    ignore_non_dom: bool,
+}
```

Comment:
You created an option, but it seems we are not using it. In order to "wire" an option to Biome, you have to:
1. Add it to the list of rules that have options:
https://github.com/biomejs/biome/blob/e6c12c0fed2a1b1662f2f228d369917dd58b7a5c/crates/biome_js_analyze/src/options.rs#L30-L42

https://github.com/biomejs/biome/blob/e6c12c0fed2a1b1662f2f228d369917dd58b7a5c/crates/biome_js_analyze/src/options.rs#L54

https://github.com/biomejs/biome/blob/e6c12c0fed2a1b1662f2f228d369917dd58b7a5c/crates/biome_js_analyze/src/options.rs#L155

2. You must apply all the correct macros, here's an example:
https://github.com/biomejs/biome/blob/e6c12c0fed2a1b1662f2f228d369917dd58b7a5c/crates/biome_js_analyze/src/semantic_analyzers/style/no_restricted_globals.rs#L62-L70

3. Apply `FromStr`, with no fancy implementation
https://github.com/biomejs/biome/blob/e6c12c0fed2a1b1662f2f228d369917dd58b7a5c/crates/biome_js_analyze/src/semantic_analyzers/style/no_restricted_globals.rs#L77

4. And apply our internal deserialisation
https://github.com/biomejs/biome/blob/e6c12c0fed2a1b1662f2f228d369917dd58b7a5c/crates/biome_js_analyze/src/semantic_analyzers/style/no_restricted_globals.rs#L86-L109


For testing, the guide should help you: https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#rule-configuration


Once you've done this, there are two missing things:
- add documentation comments for `allowed_invalid_roles` and `ignore_non_dom` and explain what they do
- update the documentation of this rule by adding the options and explaining them:
https://github.com/biomejs/biome/blob/e6c12c0fed2a1b1662f2f228d369917dd58b7a5c/crates/biome_js_analyze/src/semantic_analyzers/style/no_restricted_globals.rs#L35-L50